### PR TITLE
Match names in PR titles to students

### PR DESCRIPTION
**Pull request by kaidamasaki**
Look for missing students in group projects.

### CODE REVIEW

**Comment by kaidamasaki** [comment]
It looks like this has some minor bugs.  I'll need to poke at some logs before I can figure out why it's not quite working.

File: lib/github.rb [hunk 1]
```diff
@@ -88,7 +88,7 @@ def get_prs(repo_url)
     return pr_info
   end
 
-  def group_project(cohort_students, data, repo)
+  def group_project(cohort_students, data, repo, cohort_name)
     students = []
     url = contributors_url(data)
```
Comment (droberts-ada) [inline lib/github.rb]: I know you didn't write this originally, but I really don't like the parameter name `data`. Something like `github_pr_data` might be better, if you feel like making the change.

File: lib/github.rb [hunk 2]
```diff
@@ -99,10 +99,34 @@ def group_project(cohort_students, data, repo)
...
+        matches = cohort_students.select { |s| s.name =~ /#{name}/i }
+
+        # Do a word-match if we got multiple matches.
+        if matches.length > 1
+          matches = cohort_students.select { |s| s.name.downcase.split(/\s/).include?(name.downcase) }
+        end
+
+        # Add the user if the name was unambiguous.
+        contributor_usernames << matches.first.github_name if matches.length == 1
+      end
```
Comment (droberts-ada) [inline lib/github.rb]: If it wasn't, could you drop a log line? Might help with future debugging.

File: lib/github.rb [hunk 3]
```diff
@@ -99,10 +99,34 @@ def group_project(cohort_students, data, repo)
...
+      names = name_title.split(/\s?(?:(?:[&+,;]+)|(?: and )|(?: - ))\s?/)
+
+      names.each do |name|
+        # Get all the students that have this name as part of their name.
+        matches = cohort_students.select { |s| s.name =~ /#{name}/i }
+
+        # Do a word-match if we got multiple matches.
+        if matches.length > 1
+          matches = cohort_students.select { |s| s.name.downcase.split(/\s/).include?(name.downcase) }
+        end
```
Comment (droberts-ada) [inline lib/github.rb]: This is to filter out nick names (like "Kat" vs "Katherine")? What if the second select yields no results?

Comment (kaidamasaki) [reply lib/github.rb]: The second select only kicks in if we got multiple matches.  If the first yields multiple results and the second yields no results then we do nothing (and just go with our existing commit based logic).

File: lib/github.rb (hunk 2 above)
Comment (kaidamasaki) [reply lib/github.rb]: I just added a bunch of debug logging up above.

File: config/environment.rb [hunk 4]
```diff
@@ -1,5 +1,29 @@
...
 require File.expand_path('../application', __FILE__)
 
+# Initialize logging
+Rails.logger = Logger.new(STDOUT)
+
+Rails.logger.level = Logger::INFO # Default to INFO.
+
+# Downcase the log level only if we've got one.
+log_level = ENV['LOG_LEVEL']
+log_level &&= ENV['LOG_LEVEL'].downcase
```
Comment (droberts-ada) [inline config/environment.rb]: `&&=`!!!!!

**Pull request merged**